static/js/services/generate-search-requests.js

```diff
@@ -17,34 +17,40 @@ var _ = require('underscore'),
         });
       };
 
-      var getViewForMultidropdown = function(view, filter, mapKeys) {
-        if (!filter || !filter.selected) {
+      // filter = { selected: [...], options: [...]}
+      var getRequestForMultidropdown = function(view, filter, mapKeysFunc) {
+        var nothingSelected = !filter || !filter.selected || filter.selected.length === 0;
+        if (nothingSelected) {
           return;
         }
-        if (filter.selected.length > 0 &&
-           (!filter.options || filter.selected.length < filter.options.length)) {
 
+        // If everything is selected, no filter to apply.
+        var everythingSelected = filter.options && (filter.selected.length === filter.options.length);
+
+        // tmp in which case is there no options?? Why do we still filter?
```

Comment:
Ok, yeah select2 search uses this function with something like :
```
{
   selected: [ 'person', 'clinic' ]
}
```
instead of
```
{
   selected: [ 'person', 'clinic' ],
   options: [ 'person', 'clinic', 'district_hospital' ]
}
```
Which means this function is called multidropdown, but used for other things. So I've split out the select2Search case from the multidropdown case.

Also added a bucha tests for this function.